### ISSUE TITLE
CU launch script not written correctly by popen executing component

### ISSUE THREAD

**Issue by iparask** [radical-cybertools/radical.pilot#1201]
While running example 08_unit_environment.py, with the latest release (0.44), the output produced was:

```
unit.000127: Done, exit:   0, out: unit.000127 greets jabberwockyexpor
```
The unit launch script looks like:
```
#!/bin/sh


# Change to working directory for unit
cd /home/giannis/radical.pilot.sandbox/rp.session.giannis-VirtualBox.giannis.017204.0003-pilot.0000/unit.000004
# Environment variables
 TEST=jabberwockyexport RP_SESSION_ID=rp.session.giannis-VirtualBox.giannis.017204.0003
export RP_PILOT_ID=pilot.0000
export RP_AGENT_ID=agent_0
export RP_SPAWNER_ID=agent_0.AgentExecutingComponent.0.child
export RP_UNIT_ID=unit.000004

# The command to run
/bin/echo "$RP_UNIT_ID greets $TEST" 
RETVAL=$?
# Exit the script with the return code from the command
exit $RETVAL
```

I think it is because a space is missing [here](https://github.com/radical-cybertools/radical.pilot/blob/master/src/radical/pilot/agent/executing/popen.py#L240) after the second %s.

**Comment by vivek-bala** [comment]
You mean a "\n" is missing at the end of that line you linked? Agree.

Also since these are environment variables assignments. Shouldn't the lines read

```
for key,val in cu['description']['environment'].iteritems():
    env_string += 'export %s=%s\n' % (key, val)
```

instead of 

```
for key,val in cu['description']['environment'].iteritems():
    env_string += ' %s=%s' % (key, val)
```

**Comment by iparask** [comment]
Yes, my bad!

**Comment by marksantcroos** [comment]
> ```
> for key,val in cu['description']['environment'].iteritems():
>     env_string += 'export %s=%s\n' % (key, val)
> ```

Agreed. Please test and create a PR.

(Looking at the changelog of this piece of code hurts ones eyes ... )

**Comment by andre-merzky** [comment]
Please, while you are on it, add quotes to the key/val expansion:

```
for key,val in cu['description']['environment'].iteritems():
    env_string += 'export "%s=%s"\n' % (key, val)
```

Thanks, Andre.

**Comment by andre-merzky** [comment]
This was closed via #1203
